Introduce resource digest #1103 (#1123)
httpwg/http-extensions: draft-ietf-httpbis-digest-headers.md

The definition "validator" in this document is to be interpreted as described in
Section 7.2 of [RFC7231].

# Representation Digest {#representation-digest}

The representation digest is an integrity mechanism for HTTP resources that
decouples the checksum calculation from:

- the payload body - which may be altered by mechanism like Range Requests
  [RFC7233] or the method (eg. HEAD);

- and the message body - which depends on `Transfer-Encoding` and whatever
  transformations the intermediaries may apply.

and takes into account that `representation data` defined in  [RFC7231] as

~~~
   representation-data := Content-Encoding( Content-Type( bits ) )
~~~

can be:

- fully contained in the message body,
- partially-contained in the message body,
- or not at all contained in the message body.

{{resource-representation}} contains many examples

> showing the HTTP semantics impact on the message and payload body.

Reviewer: This still reads a bit disjointed to me, so how about something like the below, although it maybe has some errors

" The representation digest is an integrity mechanism for HTTP resources which uses a checksum that is calculated independently of the payload body and message body. It is calculated using the representation data (see [RFC7231]), can be fully or partially contained in the message body, or not contained at all:

~~~
   representation-data := Content-Encoding( Content-Type( bits ) )
~~~

It is prudent to consider the effect HTTP semantics in relation to resources. For example, the message body can be affected by range-requests or different methods such as HEAD, while the payload body is dependent on Transfer encoding or other transformations. {{resource-representation}} contains several examples to help illustrate the effects."

Author: Sounds good, I'll update the PR tomorrow! Thanks being so patient with a non-native speaker!

# Representation Digest {#representation-digest}

The representation digest is an integrity mechanism for HTTP resources
which uses a checksum  that is calculated independently of the payload body and message body.
It is calculated using the representation data (see [RFC7231]),
that can be fully or partially contained in the message body, or not contained at all:

~~~
   representation-data := Content-Encoding( Content-Type( bits ) )
~~~

> It is prudent to consider the effect of the HTTP semantics on the messages;

Author: ```suggestion
This takes into account the effect of the HTTP semantics on the messages;
```
@LPardue We are not giving implementors the freedom to be imprudent.
